middleware/setupDB.js

```diff
@@ -22,6 +22,8 @@ export const setupDB = async (app) => {
     console.log('Database initialized and middleware set up.');
   } catch (error) {
     console.error('Failed to initialize database:', error);
-    process.exit(1);
+    app.use((req, res, next) => {
```

Comment:
Just thinking about that fact we have two errors now. 

`process.exit(1)` Is a hard fail and stops the app. Where as what you've put allows the app to keep going if the database fails (Which is good in my view).

in `app.js` where we call setupDB it has an error wrapping a lot of our set up. Does this need to change now you've altered the code?

Reply:
Line `126` is where we wrap our start up code that triggers and error if all else fails.

Reply:
@psweeting1 yeah I hadn't spotted that. You're right, by failing gracefully in `setupDB.js` means that, line `126` in `app.js` would never fire. 

If we are happy for the app to stop/crash when it can't initialise the database in `setupDB.js` then I'll revert the change I've made. 

Maybe there is a nicer way to handle it in `app.js`??